crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -576,24 +579,35 @@ async fn validate_and_publish_software_list(
     Ok(vec![])
 }
 
-async fn execute_operation(payload: &str, command: &str) -> Result<(), CumulocityMapperError> {
+async fn execute_operation(
+    payload: &str,
+    command: &str,
+    operation_name: &str,
+) -> Result<(), CumulocityMapperError> {
     let command = command.to_owned();
     let payload = payload.to_string();
 
-    let _handle = tokio::spawn(async move {
-        let mut child = tokio::process::Command::new(command)
-            .args(&[payload])
-            .stdin(Stdio::null())
-            .stdout(Stdio::null())
-            .stderr(Stdio::null())
-            .spawn()
-            .map_err(|e| CumulocityMapperError::ExecuteFailed(e.to_string()))
-            .unwrap();
-
-        child.wait().await
-    });
+    let child = tokio::process::Command::new(&command)
+        .args(&[&payload])
+        .stdin(Stdio::null())
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()
+        .map_err(|e| CumulocityMapperError::ExecuteFailed {
+            error_message: e.to_string(),
+            command: command.to_string(),
+            operation_name: operation_name.to_string(),
+        });
 
-    Ok(())
+    match child {
+        Ok(mut child) => {
+            let _ = child.wait().await;
```

Comment:
This is where I would re-introduce the now missing `tokio::spawn`. __This has to be double-checked__.

```suggestion
            let _ = tokio::spawn(child.wait());
```

Reply:
Yes, thank you!